chain/mining.go

```diff
@@ -93,17 +95,30 @@ func (bm *BuiltinMining) BuildBlock(ctx context.Context, height uint32, chordID
 			break
 		}
 
-		txnHash := txn.Hash()
-		if DefaultLedger.Store.IsTxHashDuplicate(txnHash) {
-			log.Warning("it's a duplicate transaction")
+		if err := VerifyTransaction(txn); err != nil {
+			log.Warning("invalid transaction")
+			txnCollection.Pop()
+			continue
+		}
+		if err := VerifyTransactionWithLedger(txn); err != nil {
+			log.Warning("invalid transaction")
+			txnCollection.Pop()
+			continue
+		}
+		if err := bvs.VerifyTransactionWithBlock(txn, height); err != nil {
+			bvs.Reset()
+			log.Warning("invalid transaction")
```

Comment:
It's better to use `log.Warningf("invalid transaction: %v", err)` here and in L104